File: backend/src/main/kotlin/fr/gouv/dgampa/rapportnav/infrastructure/api/bff/adapters/v2/MissionActionInput.kt
package fr.gouv.dgampa.rapportnav.infrastructure.api.bff.adapters.v2

import fr.gouv.dgampa.rapportnav.domain.entities.mission.env.MissionSourceEnum
import fr.gouv.dgampa.rapportnav.domain.entities.mission.nav.action.ActionType

data class MissionActionInput(
    val id: String,
    val missionId: Int,
    val actionType: ActionType,
    val source: MissionSourceEnum,
    val env: MissionEnvActionDataInput? = null,
    val nav: MissionNavActionDataInput? = null,
    val fish: MissionFishActionDataInput? = null
){
}

File: frontend/src/v2/features/common/types/mission-action-input.ts
import { ActionTypeEnum, MissionSourceEnum } from '@common/types/env-mission-types'
import { MissionEnvActionDataInput } from './mission-env-action-input'
import { MissionFishActionDataInput } from './mission-fish-action-input'
import { MissionNavActionDataInput } from './mission-nav-action-input'

export interface MissionActionInput {
  id: string
  missionId: number
  source: MissionSourceEnum
  actionType: ActionTypeEnum
  env?: MissionEnvActionDataInput
  nav?: MissionNavActionDataInput
  fish?: MissionFishActionDataInput
}
